packages/driver/src/cy/multi-domain/index.ts

```diff
@@ -189,13 +192,7 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
         // to stack up listeners on it after it's originally bound
         if (!communicator.listeners('uncaught:error').length) {
           communicator.once('uncaught:error', ({ err }) => {
-            if (err?.name === 'CypressError') {
-              // This is a Cypress error thrown from the secondary origin after the command queue has finished, do not wrap it as a spec or app error.
-              cy.fail(err, { async: true })
-            } else {
-              // @ts-ignore
-              Cypress.runner.onSpecError('error')({ error: err })
-            }
```

Comment:
As far I can tell, calling `onSpecError` isn't necessary here and it just causes errors to get double-wrapped in certain circumstances. Without calling `onSpecError`, the errors still get wrapped properly (and all the tests pass with this change).

Reply:
@mjhenkes and I spent some time on this, but I can't remember why 😕 .